src/software/ai/navigator/path_planner/simulated_theta_star_test.cpp

```diff
@@ -0,0 +1,242 @@
+#include "software/simulated_tests/terminating_validation_functions/robot_in_polygon_validation.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/test_util/test_util.h"
+#include "software/time/duration.h"
+#include "software/world/world.h"
+
+class SimulatedThetaStarTest : public SimulatedTacticTestFixture
+{
+};
+
+TEST_F(SimulatedThetaStarTest, test_theta_star_robot_and_dest_in_obstacle)
+{
+    // Both initial_position and destination are in obstacle and in the same coordinate
+    Point destination      = Point(1, 0);
+    Point initial_position = Point(0.96905413818359376, -0.26277551269531252);
+    setBallState(BallState(Point(0, -0.6), Vector(0, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 0), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId({Point(1, 0)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            Rectangle expected_final_position(Point(0.985, -0.26), Point(1.15, -0.3));
+            robotInPolygon(1, expected_final_position, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(15));
+}
+
+TEST_F(SimulatedThetaStarTest, test_theta_star_dest_in_obstacle)
+{
+    // Destination is in obstacle, but initial point is open
+    Point destination      = Point(1, -0.1);
+    Point initial_position = Point(1, 2);
+    setBallState(BallState(Point(0, -0.6), Vector(0, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 0), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId({Point(1, 0)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            Rectangle expected_final_position(Point(0.985, -0.26), Point(1.15, -0.3));
```

Comment:
Comment added

Reply:
Why this rectangle? Why doesn't this rectangle contain the destination (1,0)?

Reply:
Since the enemy robot is at (1, 0) and contains the destination with in it, the friendly robot avoids ramming into the enemy robot by going to the nearest free point which ends up being below the enemy robot.